src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -198,13 +198,30 @@ private async Task UpdateSymbolTreeInfoAsync(Project project, CancellationToken
                 Debug.Assert(SupportAnalysis(project));
 
                 // Produce the indices for the source and metadata symbols in parallel.
-                var tasks = new List<Task>
+                var isRemoteWorkspace = project.Solution.Workspace.Kind == WorkspaceKind.RemoteWorkspace;
+                Task[] tasks = isRemoteWorkspace
+                    ? UpdateSymbolTreeInfoInServiceProcessAsync(project, cancellationToken)
+                    : UpdateSymbolTreeInfoInCurrentProcessAsync(project, cancellationToken);
+
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+            }
+
+            private Task[] UpdateSymbolTreeInfoInCurrentProcessAsync(Project project, CancellationToken cancellationToken)
```

Comment:
:bulb: Use a value tuple here to avoid the array allocation

Reply:
Do you have a tupletaskawaiter?

Reply:
@Therzok https://www.nuget.org/packages/TaskTupleAwaiter/ or [https://gist.github.com/jnm2/3660db29457d391a34151f764bfe6ef7](https://gist.github.com/jnm2/3660db29457d391a34151f764bfe6ef7)?

Reply:
As commented [here](https://github.com/dotnet/roslyn/pull/33173/files#r271560126), we don't gain anything by using ValueTuple, as Task.WhenAll is necessary. Thus, we'll end up allocating an array _either way_. So keeping the array.

Reply:
> As commented here, we don't gain anything by using ValueTuple, as Task.WhenAll is necessary

We should revisit this claim after the other issue is resolved.